### PR TITLE
chore(ci): Migrate CI jobs to circleci

### PR DESCRIPTION
**Pull request by rpl**
- migrate CI jobs from travis to circleci

TODO:
- [x] enable circleci on this repo (either by enabling it on the mozilla-jetpack organization or by moving this repo under the mozilla organization)
- [x] setup NPM_TOKEN on the circleci project
- [x] double-check project settings match the one used for other github repos (e.g. web-ext circleci settings)

### CODE REVIEW

**Comment by rpl** [comment]
@muffinresearch would you mind to setup circleci for this repo? (and regenerate and set on the circleci side the npm token as we did for the other repos).

**Comment by muffinresearch** [comment]
@rpl I don't have admin on this repo. If you do, you should be able to enable here:  https://app.circleci.com/projects/project-setup/github/mozilla-jetpack/node-fx-runner. 

If this doesn't have anyone with admin, we'll need to ask an org admin for assistance. We might want to consider moving this project to the mozilla org too at some point - though that's a separate issue.

**Comment by rpl** [comment]
@muffinresearch it looks that I can't setup a circleci project on this repo neither (I guess that it may be because circleci has to be enabled at an organization level and it is not unlikely that it has never been enabled before on the mozilla-jetpack organization).

I'm thinking that it may be more reasonable to ask someone with the required permissions on the mozilla-jetpack and mozilla organizations to move this repository under the mozilla organization, instead of enabling circleci on this organization just for this repo (which we would want to move in any case). 

what do you think?

**Comment by muffinresearch** [comment]
@rpl Yep, that sounds like a plan.

**Comment by muffinresearch** [comment]
Access issues resolved and token has been added.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  restore_build_cache:
+    description: restore npm package cache
+    parameters:
+      suffix:
+        type: string
+        default: default
+    steps:
+      - restore_cache:
+          keys:
+            - npm-packages-{{ checksum "package-lock.json" }}-<< parameters.suffix >>
```
Comment (muffinresearch) [inline .circleci/config.yml:61]: This is cool, we could do this more everywhere else to save on duplication on cache creation/restoration.

**Review by muffinresearch — APPROVED** [review]
lgtm r+

File: .circleci/config.yml (hunk 1 above)
Comment (rpl) [reply .circleci/config.yml:61]: yep, it is, the jobs do definitely run faster thanks to it and the checksum and suffix clear my concerns about stale caches and cached node_modules dirs built with different npm major versions.

If I'm not mistaken we are already using this trick in all the repository where we do have a lock file (and I'm 100% sure that we use it for the addons-linter and web-ext repositories).

**Pull request merged**